libsolidity/ast/Types.h

```diff
@@ -1205,6 +1205,7 @@ class MappingType: public Type
 	{
 		return _inLibrary ? shared_from_this() : TypePointer();
 	}
+	virtual unsigned calldataEncodedSize(bool) const override { return 32; }
```

Comment:
Hm... This is pretty hacky now...

Reply:
This should not be changed. You should go via `encodingType()`

Reply:
Actually I don't think any change in Types.h is needed.

Reply:
Some change is needed either here or in code generation (code generation has to know that it can pass down the storage base reference for external library calls that return mappings - without this change code generation will currently throw here: https://github.com/ethereum/solidity/blob/b7a23ee1699f93c8ae8307783142ca96e5df060f/libsolidity/codegen/ExpressionCompiler.cpp#L2091).

Maybe it's better to change https://github.com/ethereum/solidity/blob/b7a23ee1699f93c8ae8307783142ca96e5df060f/libsolidity/codegen/ExpressionCompiler.cpp#L1873? I'm currently not sure.

Reply:
Maybe https://github.com/ethereum/solidity/blob/b7a23ee1699f93c8ae8307783142ca96e5df060f/libsolidity/codegen/ExpressionCompiler.cpp#L1873 should actually be
``retSize += retType->encodingType()->calldataEncodedSize()`` or something like that...

Reply:
That seems to work at least.

Reply:
Ok, that seems to break some other tests (strangely only on CI not locally, I'll check).